tests/unit/test_models.py

```diff
@@ -164,5 +164,20 @@ def test_invalid(self):
         )
 
 
+class TestAppData(unittest.TestCase):
+    def test_invalid(self):
+        with self.assertRaises(ValueError):
+            AppData("this is not valid")
```

Comment:
Fun fact! You can also assert the error message like this. Not sure if the string is correct.
```suggestion
        with self.assertRaises(ValueError) as err:
            AppData("this is not valid")
        self.assertEquals(str(err), 'invalid app data "this is not valid"')
```